app/scripts/metamask-controller.js

```diff
@@ -63,16 +63,19 @@ module.exports = class MetamaskController {
   }
 
   getState () {
-    return extend(
-      this.state,
-      this.ethStore.getState(),
-      this.configManager.getConfig(),
-      this.keyringController.getState(),
-      this.txManager.getState(),
-      this.noticeController.getState(), {
-        lostAccounts: this.configManager.getLostAccounts(),
-      }
-    )
+    return this.keyringController.getState()
+    .then((keyringControllerState) => {
+      return extend(
```

Comment:
out of scope for this PR but...

we should prolly keep these states in separate name spaces instead of squashing them all into a single object (could have a collision that leads to bugs)